test/box-luatest/error_subsystem_improvements_test.lua

```diff
@@ -79,3 +79,14 @@ g.test_custom_payload = function()
         custom_type = "MyAppError",
     })
 end
+
+-- Test the `message` argument to the table constructor of `box.error.new`
+-- (gh-9102).
+g.test_error_new_message_arg = function()
+    local msg = 'message'
+    t.assert_equals(box.error.new{msg}.message, msg)
+    t.assert_equals(box.error.new{message = msg}.message, msg)
+    t.assert_equals(box.error.new{reason = msg}.message, msg)
+    t.assert_equals(box.error.new{msg, message = 'other'}.message, msg)
+    t.assert_equals(box.error.new{message = msg, reason = 'other'}.message, msg)
```

Comment:
Looks like it is worth to test [1] precedence othe `reason` field.

Also it does not look safe as stated in issue. It would be safe if `reason` has the highest precedence. But may be I'm just too paranoid.

Reply:
> Looks like it is worth to test [1] precedence othe `reason` field.

Fixed.

> Also it does not look safe as stated in issue. It would be safe if reason has the highest precedence. But may be I'm just too paranoid.

I think that if the user passed a `message` field or an unnamed field prior to 3.1 (which would previously be simply ignored), it is him to blame.